src/replication/stream_senders_manager.cpp

```diff
@@ -0,0 +1,244 @@
+/*
+ * Copyright (C) 2008 Search Solution Corporation. All rights reserved by Search Solution.
+ *
+ *   This program is free software; you can redistribute it and/or modify
+ *   it under the terms of the GNU General Public License as published by
+ *   the Free Software Foundation; either version 2 of the License, or
+ *   (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program; if not, write to the Free Software
+ *  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+ *
+ */
+
+/*
+ * stream_senders_manager.cpp - 
+ */
+
+#include "stream_senders_manager.hpp"
+#include "replication_common.hpp"
+
+#include <utility>
+#include "thread_manager.hpp"
+#include "thread_daemon.hpp"
+
+/* TODO : I am not sure we will not need some other functionality specific to replication 
+ * keep this in replication namespace and folder until it is more clear */
+namespace cubreplication
+{
+
+  const unsigned int stream_senders_manager::SUPERVISOR_DAEMON_DELAY_MS = 10;
+  const unsigned int stream_senders_manager::SUPERVISOR_DAEMON_CHECK_CONN_MS = 5000;
+
+  stream_senders_manager::stream_senders_manager (cubstream::stream &supervised_stream):
+    m_stream (supervised_stream)
```

Comment:
stream_senders and supervisor_daemon are not initialized in constructor, as well please use m_ prefix for class members